incubator/mysql/templates/secrets.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+type: Opaque
+data:
+  {{ if .Values.mysqlRootPassword }}
+  mysql-root-password:  {{ .Values.mysqlRootPassword | b64enc | quote }}
+  {{ else }}
+  mysql-root-password: {{ randAlphaNum 10 | b64enc | quote }}
```

Comment:
From the docs:

>{{if pipeline}} T1 {{end}}
>	If the value of the pipeline is empty, no output is generated;
>	otherwise, T1 is executed.  The empty values are false, 0, any
>	nil pointer or interface value, and any array, slice, map, or
>	string of length zero.
>	Dot is unaffected.

I'm not sure how to test for whether it's set or not. Possibly `len(.Values.mysqlRootPassword) > 0` but that would probably throw an error if it wasn't defined?